### DISCUSSION TITLE
150 Android user test case, 35% got CharonErrorState.UNREACHABLE

### DISCUSSION THREAD

**Discussion by realbitt** [strongswan/strongswan#1523] [General]
Hi, 
I have developed android app using **strongswan** lib. I sent my app to 150 android users for testing and for most of them working as expected, but around 35% of them can't able to connect to server and they got **CharonErrorState.UNREACHABLE**

 **all who can't able to connect to my server they can :**
*  ping server domain successfully and they can see it .
* using home network or mobile network normally and no special case.

I have used strongswan native android app to see detailed log, and you can see **"NO PEER RESPONDING"** after 3 retransmits.
![image](https://user-images.githubusercontent.com/17877111/214909304-44cfbb33-df7e-4e8c-919c-90f494b9f018.png)

What is the cause of the problem, and can solve it from the server side?

**Comment by tobiasbrunner** [comment]
> I have developed android app using strongswan lib

Please be aware of [strongSwan's license](https://www.strongswan.org/license.html).

> What is the cause of the problem

Most likely the traffic is blocked by a firewall (either accidentally or deliberately).

> and can solve it from the server side?

Does the server receive these requests? Does it send a response?

What you could try is using a [custom server port](https://docs.strongswan.org/docs/5.9/features/natTraversal.html#_custom_server_ports) and configure that on the client (in the strongSwan app in the advanced profile settings).

**Reply by realbitt** [reply]
`Please be aware of [strongSwan's license](https://www.strongswan.org/license.html).`
Yes, I will.
`Most likely the traffic is blocked by a firewall (either accidentally or deliberately).`
how and why, but same user when move to another network he can connect normally.
`Does the server receive these requests? Does it send a response?`
the server receive requests from app but no response from server as i saw in log, please see attached picture in the main post.

**Reply by tobiasbrunner** [reply]
> how and why, but same user when move to another network he can connect normally.

What do you mean? Firewalls, NAT routers, routing etc. can be configured and behave completely different in different networks.

> the server receive requests from app

Are you sure? And it sends a response? Can you post the server log that corresponds to such a client's request?
